### PR TITLE
Sarah/filter

### PR DESCRIPTION
**Pull request by shughes26**
## Description
Filters for resources, added filters for skill level, cloud provider, topic, and programming language

Fixes https://github.com/pulumi/marketing/issues/822.

## Checklist:
- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [x] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [x] I have manually confirmed that all new links work.
- [x] I added aliases (i.e., redirects) for all filename changes.
- [x] If making css changes, I rebuilt the bundle.

### CODE REVIEW

**Comment by sean1588** [comment]
would there be a way for you to have the groups within each filter only allow one selection, so if you choose aws, it deselects azure? When both are selected it says there are no matches. I guess maybe another option would be to OR the filters selections together rather than AND, so it would show all workshops that are aws or azure.


Screen Shot 2024-04-10 at 2 48 24 PM

**Comment by cnunciato** [comment]
> I guess maybe another option would be to OR the filters selections together rather than AND

This is how it works in the Registry filters, too (i.e., the checkboxes are ORs rather than ANDs).

**Comment by shughes26** [comment]
@sean1588 @cnunciato I'll change the filters from AND to OR

File: themes/default/theme/src/scss/_event-list.scss
Comment (cnunciato) [inline themes/default/theme/src/scss/_event-list.scss:65]: Could this be this?

```suggestion
        @apply items-end pb-4;
```

**Review by cnunciato — COMMENTED** [review]
This looks great! I agree with Sean, though, that we probably want the filters to be ORs rather than ANDs. Otherwise, only a few tiny nits with suggestions.

File: themes/default/content/resources/getting-started-with-infrastructure-as-code-on-digital-ocean/index.md
Comment (cnunciato) [inline themes/default/content/resources/getting-started-with-infrastructure-as-code-on-digital-ocean/index.md:74]: Why is it when I choose this, I get no results? 

image

I get this with others as well, like CodeFresh in the Topics menu, Azure, etc. Is this because they're in the past, and that's preventing them from displaying?

Comment (shughes26) [reply themes/default/content/resources/getting-started-with-infrastructure-as-code-on-digital-ocean/index.md:74]: The filters are OR'd except for upcoming/on-demand (since the content lives under tabs) so the only DigitalOcean content shows up under the on-demand tab.

Comment (cnunciato) [reply themes/default/content/resources/getting-started-with-infrastructure-as-code-on-digital-ocean/index.md:74]: Oh -- so the tabs behave like a filter also, then. That seems like a design bug -- makes it look like we don't have content we actually do. 😬

We don't need to block on it, but we should file a follow-up issue for it, as I'm sure someone else will bump into it and get the wrong impression. Unless you can think of an easy solution to include in this PR?

**Review by cnunciato — APPROVED** [review]
Approving to unblock, but we should file an issue for the tabs hiding filtered content and link it to this one if we decide to go that route.

File: themes/default/content/resources/getting-started-with-infrastructure-as-code-on-digital-ocean/index.md
Comment (shughes26) [reply themes/default/content/resources/getting-started-with-infrastructure-as-code-on-digital-ocean/index.md:74]: I don't really have a solution in mind so maybe we need some UX support? I created a [follow-up issue](https://github.com/pulumi/pulumi-hugo/issues/4175) for this

**Pull request merged**
